https://github.com/MariaDB/server/pull/4170

From 31aa8b6939ee9326b4145a9cceae7e5a3711d7bf Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marko=20M=C3=A4kel=C3=A4?= <marko.makela@mariadb.com>
Date: Mon, 7 Jul 2025 09:30:34 +0300
Subject: [PATCH] MDEV-37170 Enable AVX10.1 CRC-32 on GCC 16

The AVX512 accelerated CRC-32 computation that had been added
in commit 9ec7819c585d139c8fe64d0f7f0f0f51dcafa01f was disabled
in commit a293dfd92a8cf2bf3bf279b7e68b19e6e80d6e48 for GCC 16.

Let us enable that logic by applying
dr-m/crc32_simd@075bacb0cc7da8df18afa15a827ae4d11166fe32
which makes use of the avx10.1 target attribute that had been
introduced in GCC 15.
---
 mysys/crc32/crc32c_x86.cc | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/mysys/crc32/crc32c_x86.cc b/mysys/crc32/crc32c_x86.cc
index fb5dc19f..a66093e5 100644
--- a/mysys/crc32/crc32c_x86.cc
+++ b/mysys/crc32/crc32c_x86.cc
@@ -25,6 +25,9 @@
 #else
 # include <cpuid.h>
 # ifdef __APPLE__ /* AVX512 states are not enabled in XCR0 */
+# elif __GNUC__ >= 15
+#  define TARGET "pclmul,avx10.1,vpclmulqdq"
+#  define USE_VPCLMULQDQ __attribute__((target(TARGET)))
 # elif __GNUC__ >= 14 || (defined __clang_major__ && __clang_major__ >= 18)
 #  define TARGET "pclmul,evex512,avx512f,avx512dq,avx512bw,avx512vl,vpclmulqdq"
 #  define USE_VPCLMULQDQ __attribute__((target(TARGET)))
-- 
2.49.1
